server/controllers/reports_proposed/data/account_result.js

```diff
@@ -0,0 +1,239 @@
+// reports_proposed/data/bilan.js
+// Collects and aggregates data for the enterprise bilan
+
+var q       = require('q');
+var db      = require('../../../lib/db');
+var numeral = require('numeral');
+
+var formatDollar = '$0,0.00';
+var resultAccountDate = new Date();
+
+// expose the http route
+exports.compile = function (options) {
+  'use strict';
+
+  var deferred = q.defer(), context = {}, infos = {}, chargeData = {}, profitData = {};
+  context.reportDate = resultAccountDate.toDateString();
+
+  var sql =
+    'SELECT `acc`.`id` AS `accountId`, `acc`.`account_txt` AS `accounTxt`, `acc`.`account_number` AS `accountNumber`, ' +
+    '`ref`.`id` AS `referenceId`, `ref`.`ref` AS `referenceAbbr`, `ref`.`text` AS `referenceLabel`, ' +
+    '`ref`.`position` AS `referencePosition`, `ref`.`is_report` AS `referenceIsReport`, ' +
+    '`src`.`id` AS `sectionResultId`, `src`.`text` AS `sectionResultLabel`, `src`.`is_charge` AS `sectionResultIsCharge`, ' +
+    '`src`.`position` AS `sectionResultPosition`, SUM(`gld`.`debit_equiv`) AS `generalLegderDebit`, SUM(`gld`.`credit_equiv`) AS `generalLegderCredit` ' +
+    'FROM `section_resultat` `src` JOIN `reference` `ref` ON `ref`.`section_resultat_id` = `src`.`id` ' +
+    'JOIN `account` `acc` ON `acc`.`reference_id` = `ref`.`id` JOIN `general_ledger` `gld` ON `gld`.`account_id` = `acc`.`id` WHERE `gld`.`trans_date`<= (SELECT MAX(`period_stop`) ' +
+    'FROM `period` WHERE `period`.`fiscal_year_id`=?) AND `acc`.`is_ohada`=? AND SUBSTRING(`acc`.`account_number`, 1, 1) >= 6 GROUP BY `gld`.`account_id` ORDER BY `src`.`position`, `ref`.`position` DESC;';
+
+  db.exec(sql, [options.fy, 1])
+  .then(function (currentAccountDetails) {
+    infos.currentAccountDetails = currentAccountDetails || [];
+    return db.exec(sql, [options.pfy, 1]);
+  })
+  .then(function (previousAccountDetails){
+    infos.previousAccountDetails = previousAccountDetails || [];
+    return q.when(infos);
+  })
+  .then(function (infos){
+    //data processing
+
+    var chargeGeneral = 0, chargeGeneralPrevious = 0, profitGeneral = 0, profitGeneralPrevious = 0;
+
+    chargeData.currentAccountDetails = infos.currentAccountDetails.filter(function (item){
+      return item.sectionResultIsCharge === 1;
+    });
+
+    chargeData.previousAccountDetails = infos.previousAccountDetails.filter(function (item){
+      return item.sectionResultIsCharge === 1;
+    });
+
+    profitData.currentAccountDetails = infos.currentAccountDetails.filter(function (item){
+      return item.sectionResultIsCharge === 0;
+    });
+
+    profitData.previousAccountDetails = infos.previousAccountDetails.filter(function (item){
+      return item.sectionResultIsCharge === 0;
+    });
+
+
+    context.chargeSide = processCharge(chargeData);
+    context.profitSide = processProfit(profitData);
+
+
+    function processCharge (tbl){
```

Comment:
The functions `processCharge` and `processProfit` are almost identical.  Is there anyway to make the code into a single function?  It will be easier to maintain in the long run.